File: app/src/main/java/closer/vlllage/com/closer/ui/Animate.java
package closer.vlllage.com.closer.ui;

import android.view.View;
import android.view.animation.AlphaAnimation;
import android.view.animation.Animation;

import closer.vlllage.com.closer.pool.PoolMember;

public class Animate extends PoolMember {
    public void alpha(View view, boolean show) {
        view.clearAnimation();

        if (show) {
            if (view.getVisibility() == View.GONE) {
                AlphaAnimation animation = new AlphaAnimation(0, 1);
                animation.setDuration(195);
                view.startAnimation(animation);
                view.setVisibility(View.VISIBLE);
            } else {
                AlphaAnimation animation = new AlphaAnimation(view.getAlpha(), 1);
                animation.setDuration(195);
                view.startAnimation(animation);
            }
        } else {
            if (view.getVisibility() == View.GONE) {

            } else {
                AlphaAnimation animation = new AlphaAnimation(view.getAlpha(), 0);
                animation.setDuration(225);
                animation.setAnimationListener(new Animation.AnimationListener() {
                    @Override
                    public void onAnimationStart(Animation animation) {

                    }

                    @Override
                    public void onAnimationEnd(Animation animation) {
                        view.setVisibility(View.GONE);
                    }

                    @Override
                    public void onAnimationRepeat(Animation animation) {

                    }
                });
                view.startAnimation(animation);
            }
        }
    }
}

File: app/src/main/java/closer/vlllage/com/closer/ui/Animate.kt
package closer.vlllage.com.closer.ui

import android.view.View
import android.view.animation.AlphaAnimation
import android.view.animation.Animation
import closer.vlllage.com.closer.extensions.visible

import com.queatz.on.On

class Animate constructor(private val on: On) {
    fun alpha(view: View, show: Boolean) {
        view.clearAnimation()

        if (show) {
            if (view.visibility == View.GONE) {
                val animation = AlphaAnimation(0f, 1f)
                animation.duration = 195
                view.startAnimation(animation)
                view.visible = true
            } else {
                val animation = AlphaAnimation(view.alpha, 1f)
                animation.duration = 195
                view.startAnimation(animation)
            }
        } else {
            if (view.visibility == View.GONE) {

            } else {
                val animation = AlphaAnimation(view.alpha, 0f)
                animation.duration = 225
                animation.setAnimationListener(object : Animation.AnimationListener {
                    override fun onAnimationStart(animation: Animation) {

                    }

                    override fun onAnimationEnd(animation: Animation) {
                        view.visible = false
                    }

                    override fun onAnimationRepeat(animation: Animation) {

                    }
                })
                view.startAnimation(animation)
            }
        }
    }
}
